[DDW-587] Implement expandable Token view

daniloprates opened the pull request:
This PR implements the new expandable Token view.

## Todos

- [x] Create the expandable Asset component
- [x] Display the assets in the loading state

## Screenshots

![en1](https://user-images.githubusercontent.com/1504716/122104958-9e7f8580-cdee-11eb-8311-a736a89f5345.png)
![en2](https://user-images.githubusercontent.com/1504716/122104961-9fb0b280-cdee-11eb-9836-cde539ab499d.png)
![jp1](https://user-images.githubusercontent.com/1504716/122104963-9fb0b280-cdee-11eb-9388-843f6b7fdca0.png)
![jp2](https://user-images.githubusercontent.com/1504716/122104964-a0494900-cdee-11eb-84f2-9cb600e58c55.png)


---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test


---

##Test Scenarios

**Scenario 1 -  Validate initial native token view**
```gherkin
Given that I have a wallet with Native tokens
When I validate the "Summary" screen for this wallet
Then I can see that the tokens are displayed on a list
And each item can be expanded to display more information
And each row before expanding display the fingerprint and amount
And over the list the button "Show search" is displayed
```
**Scenario 2 -  Validate expanded native token view**
```gherkin
Given that I have a wallet with Native tokens
And I am on the "Summary" screen
When I click on a row from the list
Then I can see that the item expands 
And the metadata from this token is displayed below
And the options "Settings" and "Send" are displayed at the bottom
```
**Scenario 3 -  Access to "Native token settings"**
```gherkin
Given that I have a wallet with Native tokens
And I am on the "Summary" screen
When I click on a row from the list
And I click on "Settings"
Then I can see that the "Native token settings" dialog is displayed
```
**Scenario 4 - Send a native token from "Summary" screen**
```gherkin
Given that I have a wallet with Native tokens
And I am on the "Summary" screen
When I click on the arrow from a native token on the list
And I click on "Send"
Then I can see that the "Send" screen is displayed
And upon inserting a valid address and amount the selected native token is displayed as "Token #1"
```
**Scenario 5 - Validate a native token with recommended precision**
```gherkin
Given that I have a wallet with Native tokens
And I am on the "Summary" screen
And one of the tokens has a recommended decimal precision
And the selected precision is different from the recommended
When I validate the token on the list
Then I can see that an exclamation icon is displayed next to the fingerprint
And it is also visible on the "Settings" button once the list is expanded
```
**Scenario 6 - Validate explanation tooltip for a native token with recommended precision**
```gherkin
Given that I have wallet with Native tokens
And I am on the "Summary" screen
And one of the tokens has a recommended decimal precision
And the selected precision is different from the recommended
When I hover the exclamation icon 
Then I can see that an explanation tooltip is displayed
```
**Scenario 7 - Copy metadata from the list**
```gherkin
Given that I have wallet with Native tokens
And I am on the "Summary" screen
When I click on the arrow from a native token on the list
And I hover any item of the metadata
Then I can see a copy icon is displayed
And upon clicking it, the information is copied to the clipboard
And a notification is displayed at the top
```
**Scenario 8 -  Show search input**
```gherkin
Given that I have a synced wallet with Native tokens
When I validate the native token list on "Summary" screen for this wallet
And I click on "Show search"
Then I can see that an input is displayed over the list
And the text on the button "Show search" is replaced by "Hide search"
```
**Scenario 9 - Hide search input**
```gherkin
Given that I have a synced wallet with Native tokens
And the search input for native tokens is displayed
When I click on "Hide search"
Then I can see that an input is no longer displayed over the list
And the text on the button "Hide search" is replaced by "Show search"
```
**Scenario 10 - Search native tokens**
```gherkin
Given that I have a synced wallet with Native tokens
And the search input for native tokens is displayed
When I insert a value that partially matches any metadata from the list
Then I can see only the matching results are displayed on the list
```
**Scenario 11 - Search native tokens with no results**
```gherkin
Given that I have a synced wallet with Native tokens
And the search input for native tokens is displayed
When I insert a value that doesn't match any metadata from the list
Then I can see no item is displayed on the list
And the following message is displayed:

- English: No results matching your query
- Japanese: 該当する結果はありません
```
**Scenario 12 - Search Native Tokens and hide search input**
```gherkin
Given that I have a synced wallet with Native tokens
And the search input for native tokens is displayed
And a search was performed 
When I click on "Hide search"
Then I can see that an input is hidden
And the text on the button "Hide search" is replaced by "Show search"
And over the list the current search keywords are displayed with an X button
```
**Scenario 13 - Hide search and reopen from the keyword**
```gherkin
Given that I have a synced wallet with Native tokens
And the search input for native tokens is displayed
And a search was performed 
When I click on "Hide search"
And I click on the keyword
Then the keyword is no longer displayed over the input
And the search input is displayed again containing the keyword
And the text on the button "Show search" is replaced by "Hide search"
```
**Scenario 14 - Hide search and clear**
```gherkin
Given that I have a synced wallet with Native tokens
And the search input for native tokens is displayed
And a search was performed 
When I click on "Hide search"
And I click on the X next to the keyword
Then the search is cleared
And the keyword is no longer displayed
And the complete list of Native Tokens is displayed
```
**Scenario 15 - Search and clear**
```gherkin
Given that I have a synced wallet with Native tokens
And the search input for native tokens is displayed
And a search was performed 
When I click on the X button at the right of the input
Then I can see that the input is cleared
And the complete list of Native Tokens is displayed
```

---
## Issue summary

**Fixed**

- It would be great if we can fit the ticker no matter the amount of Tokens.
[Ticker does not fit](https://drive.google.com/file/d/1YzRXHrEf_UQXtt099y6J4WwFBtFeQo9g/view?usp=sharing). Reported [here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-864895032).

- For dropdown arrow, you are using wrong color -> it should be color of a text here. Plus -> move this arrow 1px top. Reported [here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-865022886).

- Add letter spacing here pls. Reported [here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-865038983).

- Increase "x" size. Reported [here].(https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-865039756)

- Empty search should not be allowed. Reported [here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-865040683).

- The alignment on the metadata tooltip is incorrect. Also, the fingerprint is not displayed in the same way. [This](https://drive.google.com/file/d/1S7_-2695ZjfVdNFt99CEeypIftWzHmzy/view?usp=sharing) is how it looks on develop and [this](https://drive.google.com/file/d/1G4S4e3grRl4xMOrxc-f4JstIsXrRg_Zj/view?usp=sharing) is on the latest build. Reported [here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-865058537).

- The warning icon is visible on the header of the native tokens dialog. Was this introduced in this PR? I can't see it on develop branch. Also, it is not aligned with the text. Check [this](https://drive.google.com/file/d/15ux45CmTysB2CWxU4XxJKDGLzCMYF7ul/view?usp=sharing) screenshot.

- The decimal precision can't be changed for any Native Token. Check [this video](https://drive.google.com/file/d/1rm28DUdjcK66E0GzpB2t4ETLG-nParCy/view?usp=sharing) for "Amazing Coin". [Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-863324020).

- I can't change the decimal precision of the tokens. Check [this](https://drive.google.com/file/d/1qN4FbvgLZ73SuqWnyMQKn_hFRjiO3h-x/view?usp=sharing) video. [Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-862721817).

- The arrow still overlaps the text, if there is a ticker next to the amount or if the exclamation sign is present. Check [this](https://drive.google.com/file/d/1GZbvBgI-oJOA11YolvwfsX3NZ6mBmTvm/view?usp=sharing) screenshot. [Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-862721817).

- "X" button in search input is too small. [Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-860457759).

- "BadCoin" should be regular (now light). [Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-860458765).

- "Amount" is not formatted. [Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-860460854.

- "X" on the keyword seems to be very small and not properly aligned. [Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-860464194)

- The "search" button should be moved to the right a bit. [Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-860465148)

- Reduce focus border twice (if now it's 4px -> use 2px). [Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-860470852)

- There is no bottom margin for the last item on the list. Check [this](https://drive.google.com/file/d/1a9s-l1GWibBU9soHBI0G7nlS2fb_Lg3k/view?usp=sharing) screenshot. [Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-862351553).

- In Japanese, there is an overlap on the fingerprint. Check [this](https://drive.google.com/file/d/1RTcoZT2bt1GRdWtqnQRnnR5RcS5csLoP/view?usp=sharing) screenshot.[Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-862351553).

- The "no results" message is missing the Japanese translation. Check [this](https://drive.google.com/file/d/1KBrfMGE68WOeKcdxhbVapnROg1-qDZN_/view?usp=sharing) screenshot. [Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-862351553).

- The amount is hard to see on [Dark Blue](https://drive.google.com/file/d/1vDtXXshdfTp9EFMwS0gRHebpXKlCNJOU/view?usp=sharing) and [Dark Cardano](https://drive.google.com/file/d/1VY7usdXboDNya5SO5pC6mXhJAIQypIvP/view?usp=sharing) themes. [Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-862351553).

- If the token has a long name, the arrow can overlap the amount. Check [this](https://drive.google.com/file/d/1udTuOoQlPBib6O-TBoHibavBkguiH0Px/view?usp=sharing) screenshot. [Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-862351553).

**Rejected**

- Is there any limit of characters for the description or any of the metadata items? If not, maybe we should avoid showing a scroll bar when it's too long. Check [this](https://drive.google.com/file/d/1BqzUjzC8Eo4RiBon2qil4NETH-Lw18A2/view?usp=sharing) video.

- Are we going to have a regular table header behavior? when I click title I do sorting by this title (like we do for stakepools and rewards). [Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-860461755)

- Initially I was thinking of this bg as of a hover effect for the line. what do you think if we will have this "coloring" only on hover? (both for small and big lines). [Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-860466427)

- No coloring for search. [Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-860470089)

- Sometimes, when the data doesn't fit only the copy symbol is displayed on another row and it looks a little off. Can this be improved? Check [this](https://drive.google.com/file/d/10zmZURkIO8BdX9Gysg8B6CurMr-Cz4mV/view?usp=sharing) video. [Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-862721817).

- Regarding the "copy" icon on a second row, I was referring to a solution similar to what was implemented for [the external link icon](https://drive.google.com/file/d/1QTAfhKPL6JLa3JWXucqCymiCsJBOcP_i/view?usp=sharing) (the icon is always attached to the text). However, I think just a part of the fingerprint should move to a second row, so the first row doesn't look empty as well. It would look like [this](https://drive.google.com/file/d/1WAcOdmWK1u1ukMLEpjwYz9eyKgmcF6IF/view?usp=sharing). Do you think this is doable? [Reported here](https://github.com/input-output-hk/daedalus/pull/2589#issuecomment-863324020).


---

## Review Checklist

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

alexander-rukin commented:
<img width="852" alt="image" src="https://user-images.githubusercontent.com/2539345/121855517-1a4cd580-ccfc-11eb-9057-48074b194b8f.png">

- right arrow is too bold 
- after last element there should be no separator 
- in this table view we can afford to show full fingerprint in a pill (at least no one was against this on our calls :D )

alexander-rukin commented:
<img width="602" alt="image" src="https://user-images.githubusercontent.com/2539345/121855689-4ff1be80-ccfc-11eb-842a-944daed09c27.png">
<img width="572" alt="image" src="https://user-images.githubusercontent.com/2539345/121855750-5ed87100-ccfc-11eb-9fd0-87855f1071b8.png">

"X"s are different

alexander-rukin commented:
<img width="190" alt="image" src="https://user-images.githubusercontent.com/2539345/121855951-96dfb400-ccfc-11eb-99f5-f6c9987af464.png">
"BadCoin" should be regular (now light)

alexander-rukin commented:
<img width="537" alt="image" src="https://user-images.githubusercontent.com/2539345/121856289-0190ef80-ccfd-11eb-88b1-4ea092c69d06.png">
is this ok? top number is not "formatted", I suppose it should write "< 0.01" there

alexander-rukin commented:
<img width="552" alt="image" src="https://user-images.githubusercontent.com/2539345/121856463-31d88e00-ccfd-11eb-9b41-285a0114eadc.png">
are we going to have regular table hader behaviuor? when I click title I do sorting by this title (like we do for stakepools and rewards)

alexander-rukin commented:
<img width="266" alt="image" src="https://user-images.githubusercontent.com/2539345/121856677-6f3d1b80-ccfd-11eb-8358-40c18e201d40.png">
"x" seems to be very small and not properly aligned... 


===

<img width="490" alt="image" src="https://user-images.githubusercontent.com/2539345/121856826-9d226000-ccfd-11eb-933f-50e1eca46cb1.png">
this is from  designs

alexander-rukin commented:
<img width="218" alt="image" src="https://user-images.githubusercontent.com/2539345/121857094-e8d50980-ccfd-11eb-9981-70afe1fbd4d9.png">
<img width="202" alt="image" src="https://user-images.githubusercontent.com/2539345/121857115-ed012700-ccfd-11eb-888a-405a9d1917a3.png">

"search" button should be moved to the right a bit

alexander-rukin commented:
<img width="565" alt="image" src="https://user-images.githubusercontent.com/2539345/121857295-22a61000-ccfe-11eb-9456-d2f1b8a4c781.png">
initially I was thinking of this bg as of a hover effect for the line. 
what do you think if we will have this "coloring" only on hover? (both for small and big lines)

alexander-rukin commented:
<img width="216" alt="image" src="https://user-images.githubusercontent.com/2539345/121858139-f8088700-ccfe-11eb-86c7-e7081acd5b3f.png">
 no coloring for search :(

alexander-rukin commented:
<img width="116" alt="image" src="https://user-images.githubusercontent.com/2539345/121858237-17071900-ccff-11eb-8e2d-1f0a4e220e9d.png">
pls reduce focus border twice (if now it's 4px -> use 2px)

daniloprates commented:
> <img alt="image" width="190" src="https://user-images.githubusercontent.com/2539345/121855951-96dfb400-ccfc-11eb-99f5-f6c9987af464.png">
> 
> "BadCoin" should be regular (now light)

This is a component used in other places. Should it be regular all across?

daniloprates commented:
> <img alt="image" width="552" src="https://user-images.githubusercontent.com/2539345/121856463-31d88e00-ccfd-11eb-9b41-285a0114eadc.png">
> 
> are we going to have a regular table header behavior? when I click title I do sorting by this title (like we do for stakepools and rewards)

Not that I know, didn't think it was in the scope of this task

alexander-rukin commented:
> > <img alt="image" width="190" src="https://user-images.githubusercontent.com/2539345/121855951-96dfb400-ccfc-11eb-99f5-f6c9987af464.png">
> > "BadCoin" should be regular (now light)
> 
> This is a component used in other places. Should it be regular all across?

in designs all regular

daniloprates commented:
> <img alt="image" width="565" src="https://user-images.githubusercontent.com/2539345/121857295-22a61000-ccfe-11eb-9456-d2f1b8a4c781.png">
> 
> initially I was thinking of this bg as of a hover effect for the line. what do you think if we will have this "coloring" only on hover? (both for small and big lines)

I don't see a point for a color on hover, when expanded. There are distinct areas with different functions. I'd keep as it is or only in the header, when hovering

daniloprates commented:
> <img alt="image" width="216" src="https://user-images.githubusercontent.com/2539345/121858139-f8088700-ccfe-11eb-86c7-e7081acd5b3f.png">
> 
> no coloring for search :(

Not feasible. This is super complex and can't be easily done in this context

daniloprates commented:
@alexander-rukin apart from my comments above, everything else is fixed :) could you go over again?

gabriela-ponce commented:
@daniloprates Here are a couple of issues for build 18331:

- There is no bottom margin for the last item on the list. Check [this](https://drive.google.com/file/d/1a9s-l1GWibBU9soHBI0G7nlS2fb_Lg3k/view?usp=sharing) screenshot.
- In Japanese, there is an overlap on the fingerprint. Check [this](https://drive.google.com/file/d/1RTcoZT2bt1GRdWtqnQRnnR5RcS5csLoP/view?usp=sharing) screenshot.
- The "no results" message is missing the Japanese translation. Check [this](https://drive.google.com/file/d/1KBrfMGE68WOeKcdxhbVapnROg1-qDZN_/view?usp=sharing) screenshot.
- The amount is hard to see on [Dark Blue](https://drive.google.com/file/d/1vDtXXshdfTp9EFMwS0gRHebpXKlCNJOU/view?usp=sharing) and [Dark Cardano](https://drive.google.com/file/d/1VY7usdXboDNya5SO5pC6mXhJAIQypIvP/view?usp=sharing) themes.

daniloprates commented:
Thanks @gabriela-ponce. I've fixed them, except the No Results in JP, which was [already done](https://user-images.githubusercontent.com/1504716/122287211-14efb680-cec7-11eb-8844-286e85b31f7a.png) 🤔 

@alexander-rukin I've reverted the fingerprint for the short version due to [this](https://drive.google.com/file/d/1udTuOoQlPBib6O-TBoHibavBkguiH0Px/view?usp=sharing)

> @daniloprates Here are a couple of issues for build 18331:
> 
> * There is no bottom margin for the last item on the list. Check [this](https://drive.google.com/file/d/1a9s-l1GWibBU9soHBI0G7nlS2fb_Lg3k/view?usp=sharing) screenshot.
> * In Japanese, there is an overlap on the fingerprint. Check [this](https://drive.google.com/file/d/1RTcoZT2bt1GRdWtqnQRnnR5RcS5csLoP/view?usp=sharing) screenshot.
> * The "no results" message is missing the Japanese translation. Check [this](https://drive.google.com/file/d/1KBrfMGE68WOeKcdxhbVapnROg1-qDZN_/view?usp=sharing) screenshot.
> * The amount is hard to see on [Dark Blue](https://drive.google.com/file/d/1vDtXXshdfTp9EFMwS0gRHebpXKlCNJOU/view?usp=sharing) and [Dark Cardano](https://drive.google.com/file/d/1VY7usdXboDNya5SO5pC6mXhJAIQypIvP/view?usp=sharing) themes.
> * If the token has a long name, the arrow can overlap the amount. Check [this](https://drive.google.com/file/d/1udTuOoQlPBib6O-TBoHibavBkguiH0Px/view?usp=sharing) screenshot.
> 
> Note: I was able to reproduce these issues also in the latest build (18337).

gabriela-ponce commented:
@daniloprates Almost all the issues mentioned are fixed. Here's a summary of issues on the latest build:

- The arrow still overlaps the text, if there is a ticker next to the amount or if the exclamation sign is present. Check [this](https://drive.google.com/file/d/1GZbvBgI-oJOA11YolvwfsX3NZ6mBmTvm/view?usp=sharing) screenshot.
- I can't change the decimal precision of the tokens. Check [this](https://drive.google.com/file/d/1qN4FbvgLZ73SuqWnyMQKn_hFRjiO3h-x/view?usp=sharing) video.
- Sometimes, when the data doesn't fit only the copy symbol is displayed on another row and it looks a little off. Can this be improved? Check [this](https://drive.google.com/file/d/10zmZURkIO8BdX9Gysg8B6CurMr-Cz4mV/view?usp=sharing) video.

daniloprates commented:
Thanks @gabriela-ponce. Please check:


> @daniloprates Almost all the issues mentioned are fixed. Here's a summary of issues on the latest build:
> 
> * The arrow still overlaps the text, if there is a ticker next to the amount or if the exclamation sign is present. Check [this](https://drive.google.com/file/d/1GZbvBgI-oJOA11YolvwfsX3NZ6mBmTvm/view?usp=sharing) screenshot.

Fixed

> * I can't change the decimal precision of the tokens. Check [this](https://drive.google.com/file/d/1qN4FbvgLZ73SuqWnyMQKn_hFRjiO3h-x/view?usp=sharing) video.

I forgot to mention that I've hardcoded the decimals and recommended decimals for the Good and Bad coins, for testing purposes

> * Sometimes, when the data doesn't fit only the copy symbol is displayed on another row and it looks a little off. Can this be improved? Check [this](https://drive.google.com/file/d/10zmZURkIO8BdX9Gysg8B6CurMr-Cz4mV/view?usp=sharing) video.

This can't be improved, at the moment

gabriela-ponce commented:
@daniloprates This issue is fixed:

> The arrow still overlaps the text, if there is a ticker next to the amount or if the exclamation sign is present. Check [this](https://drive.google.com/file/d/1GZbvBgI-oJOA11YolvwfsX3NZ6mBmTvm/view?usp=sharing) screenshot.

A few additional comments about the other issues:

- The decimal precision can't be changed for any Native Token. Check [this video](https://drive.google.com/file/d/1rm28DUdjcK66E0GzpB2t4ETLG-nParCy/view?usp=sharing) for "Amazing Coin"
- Regarding the "copy" icon on a second row, I was referring to a solution similar to what was implemented for [the external link icon](https://drive.google.com/file/d/1QTAfhKPL6JLa3JWXucqCymiCsJBOcP_i/view?usp=sharing) (the icon is always attached to the text). However, I think just a part of the fingerprint should move to a second row, so the first row doesn't look empty as well. It would look like [this](https://drive.google.com/file/d/1WAcOdmWK1u1ukMLEpjwYz9eyKgmcF6IF/view?usp=sharing). Do you think this is doable?

I'll continue testing this build and add any additional issue on another comment 👍

DominikGuzei commented on source/renderer/app/components/assets/AssetContent.js:
```diff
@@ -0,0 +1,245 @@
+};
+
+@observer
+export default class AssetContent extends Component<Props, State> {
```
Let's write any "new" component with the latest React best practices (pure function + hooks)

daniloprates replied:
@DominikGuzei are we creating this kind of components already? It would be good to have some definitions, so we can keep it consistent.

daniloprates replied:
Done ✅

DominikGuzei commented on source/renderer/app/components/wallet/summary/WalletSummaryAsset.js:
```diff
@@ -0,0 +1,213 @@
+};
+
+@observer
+export default class WalletSummaryAsset extends Component<Props, State> {
```
Same here, function component could be used instead

daniloprates replied:
✅

DominikGuzei commented on source/renderer/app/components/wallet/summary/WalletSummaryAsset.js:
```diff
@@ -0,0 +1,213 @@
+    }));
+  };
+
+  renderHeader = () => {
```
if we are splitting up parts of the component into separate functions because it gets too large otherwise - that's a good indication for creating more, smaller components that can be composed. E.g `WalletSummaryAssetHeader` etc.

daniloprates replied:
I definitely agree we need to fragment more our components, as there are some monsters there, but not sure I agree it's the case, here. It's a very concise and interconnected component and its parts won't be used anywhere else. Don't see how it'd be beneficial to break it up into multiple files 🤔

daniloprates replied:
I've tried splitting up, but it honestly looked so messy, as the components were not meant to be used outside WalletSummaryAsset.js. I would either keep them inside or create a different naming/folder structure, so it would be clear they are internal components

DominikGuzei replied:
Ok, yeah we need to find a nicer way to write these big components … i agree that splitting them up not necessarily solves the issue - especially if these components are only used once for this case

DominikGuzei commented on source/renderer/app/containers/wallet/WalletSummaryPage.js:
```diff
@@ -114,7 +114,27 @@ export default class WalletSummaryPage extends Component<Props> {
     const walletTokens = wallet.assets.total;
-    const assetTokens = getAssetTokens(walletTokens, getAsset);
+    const assetTokens = getAssetTokens(walletTokens, getAsset)
+      /* TODO REMOVE - Begin */
```
What about this "remove" comment?

DominikGuzei reviewed (changes requested):
Great work 🎉  I have only checked the code, not tested the UX, etc.
I added a few comments and suggestions 🙏

daniloprates commented:
thanks @gabriela-ponce:

> * The decimal precision can't be changed for any Native Token. Check [this video](https://drive.google.com/file/d/1rm28DUdjcK66E0GzpB2t4ETLG-nParCy/view?usp=sharing) for "Amazing Coin"

It was due to the hardcoded bit. It was removed

> * Regarding the "copy" icon on a second row, I was referring to a solution similar to what was implemented for [the external link icon](https://drive.google.com/file/d/1QTAfhKPL6JLa3JWXucqCymiCsJBOcP_i/view?usp=sharing) (the icon is always attached to the text). However, I think just a part of the fingerprint should move to a second row, so the first row doesn't look empty as well. It would look like [this](https://drive.google.com/file/d/1WAcOdmWK1u1ukMLEpjwYz9eyKgmcF6IF/view?usp=sharing). Do you think this is doable?

This is a very complicated solution, done here because it's the same icon used throughout the app

alexander-rukin commented:
> @alexander-rukin I've reverted the fingerprint for the short version due to [this](https://drive.google.com/file/d/1udTuOoQlPBib6O-TBoHibavBkguiH0Px/view?usp=sharing)

this looks like not an issue of big "fingerprint", more like wrong trimming of "name"

daniloprates commented:
> > @alexander-rukin I've reverted the fingerprint for the short version due to [this](https://drive.google.com/file/d/1udTuOoQlPBib6O-TBoHibavBkguiH0Px/view?usp=sharing)
> 
> this looks like not an issue of big "fingerprint", more like wrong trimming of "name"

The trimming is fixed, but I think the fingerprint steals the space that could be used to the name. Anyways, long pringerprint is re-enabled

![image](https://user-images.githubusercontent.com/1504716/122571575-3fef1d00-d023-11eb-9b58-a9374af838af.png)

gabriela-ponce commented:
@daniloprates The fixes look good. I'll update the issue summary on this PR.

Here are a few minor issues/potential improvements that I found:

- The warning icon is visible on the header of the native tokens dialog. Was this introduced in this PR? I can't see it on develop branch. Also, it is not aligned with the text. Check [this](https://drive.google.com/file/d/15ux45CmTysB2CWxU4XxJKDGLzCMYF7ul/view?usp=sharing) screenshot.
- Is there any limit of characters for the description or any of the metadata items? If not, maybe we should avoid showing a scroll bar when it's too long. Check [this](https://drive.google.com/file/d/1BqzUjzC8Eo4RiBon2qil4NETH-Lw18A2/view?usp=sharing) video.

daniloprates commented:
Thanks @gabriela-ponce

> * The warning icon is visible on the header of the native tokens dialog. Was this introduced in this PR? I can't see it on develop branch. Also, it is not aligned with the text. Check [this](https://drive.google.com/file/d/15ux45CmTysB2CWxU4XxJKDGLzCMYF7ul/view?usp=sharing) screenshot.

I've fixed it, so it doesn't show in any other Asset pill, only in the Summary header

> * Is there any limit of characters for the description or any of the metadata items? If not, maybe we should avoid showing a scroll bar when it's too long. Check [this](https://drive.google.com/file/d/1BqzUjzC8Eo4RiBon2qil4NETH-Lw18A2/view?usp=sharing) video.

This is how it's design. Any asset field has a max-height of 48px and gets an scroll, beyond that.

dmitrii-gaico commented:
@daniloprates Hello!
It would be great if we can fit the ticker no matter the amount of Tokens.
[Ticker does not fit](https://drive.google.com/file/d/1YzRXHrEf_UQXtt099y6J4WwFBtFeQo9g/view?usp=sharing)
cc @alexander-rukin

alexander-rukin commented:
<img width="93" alt="image" src="https://user-images.githubusercontent.com/2539345/122767708-979fb980-d2ab-11eb-9a86-78b92cccffbb.png">
here you are using wrong color -> it should be color of a text here

alexander-rukin commented:
<img width="107" alt="image" src="https://user-images.githubusercontent.com/2539345/122770441-1dbcff80-d2ae-11eb-9195-d4d5bf32f74e.png">
add letter spacing here pls

alexander-rukin commented:
https://zpl.io/2v3ZrZn - update header to indicate clickable area

alexander-rukin commented:
<img width="156" alt="image" src="https://user-images.githubusercontent.com/2539345/122771222-e733b480-d2ae-11eb-80cc-21f2e50def66.png">
 increase "x" size

alexander-rukin commented:
<img width="167" alt="image" src="https://user-images.githubusercontent.com/2539345/122771408-12b69f00-d2af-11eb-8649-b2590a3c816a.png">
empty search should not be allowed

gabriela-ponce commented:
On build 18367:

**New**:
- The alignment on the metadata tooltip is incorrect. Also, the fingerprint is not displayed in the same way. [This](https://drive.google.com/file/d/1S7_-2695ZjfVdNFt99CEeypIftWzHmzy/view?usp=sharing) is how it looks on develop and [this](https://drive.google.com/file/d/1G4S4e3grRl4xMOrxc-f4JstIsXrRg_Zj/view?usp=sharing) is on the latest build.

**Fixed**:
>  The warning icon is visible on the header of the native tokens dialog. Was this introduced in this PR? I can't see it on develop branch. Also, it is not aligned with the text. Check [this](https://drive.google.com/file/d/15ux45CmTysB2CWxU4XxJKDGLzCMYF7ul/view?usp=sharing) screenshot.

Updating the issue summary with the latest issues ⏳

daniloprates commented:
@alexander-rukin @gabriela-ponce all fixed ✨  could you give another look, please?

Fixing now what @dmitrii-gaico reported.

daniloprates commented:
@alexander-rukin @gabriela-ponce @dmitrii-gaico all fixed ✨  could you give another look, please?

alexander-rukin commented:
> <img alt="image" width="167" src="https://user-images.githubusercontent.com/2539345/122771408-12b69f00-d2af-11eb-8649-b2590a3c816a.png">
> 
> empty search should not be allowed

@daniloprates I still can hide  a search and it will be appliend, but now no indication of it 😓 (try 3 spaces "   ")

daniloprates commented:
> > <img alt="image" width="167" src="https://user-images.githubusercontent.com/2539345/122771408-12b69f00-d2af-11eb-8649-b2590a3c816a.png">
> > empty search should not be allowed
> 
> @daniloprates I still can hide a search and it will be applied, but now no indication of it 😓 (try 3 spaces " ")
> p.s. rest seems ok for me

@alexander-rukin that's odd. I'm trimming the search value and it looks ok, to me:

![image](https://user-images.githubusercontent.com/1504716/122787716-fc4b1c00-d28b-11eb-8a91-296dd21f977b.png)

Do you Russian guys have a special space character? 😅

daniloprates commented:
@alexander-rukin @gabriela-ponce @dmitrii-gaico just pushed a new change so it won't search for assets with only spaces in the search input

gabriela-ponce commented:
@daniloprates The last fixes look good, great job. Just a small issue on the latest build:

- When the language is Japanese, the window is on the minimum supported size and a long search was performed, the "Show search" button does not fit on the screen and the x to clear the search is not visible. Check [this](https://drive.google.com/file/d/1PkI2HRcq5uS_i55lqZG73s5KHkJZ27f_/view?usp=sharing) screenshot.

daniloprates commented:
> @daniloprates The last fixes look good, great job. Just a small issue on the latest build:
> 
> * When the language is Japanese, the window is on the minimum supported size and a long search was performed, the "Show search" button does not fit on the screen and the x to clear the search is not visible. Check [this](https://drive.google.com/file/d/1PkI2HRcq5uS_i55lqZG73s5KHkJZ27f_/view?usp=sharing) screenshot.

Thanks @gabriela-ponce, nice catch. That's fixed.

gabriela-ponce reviewed (approved):
Looks good 🎉
